File: BmpGenislet.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
BMP dosyasının genişliğini 1 piksel artırır ve sağ tarafa mavi (R:0, G:0, B:255) piksel ekler
Windows 7 32-bit için uyumludur
Python 3.8+ gerektirir
"""

import sys
import os
from PIL import Image

def genislet_bmp(dosya_yolu):
    """
    BMP dosyasının genişliğini 1 piksel artırır ve sağ tarafa mavi piksel ekler
    
    Args:
        dosya_yolu: İşlenecek BMP dosyasının yolu
        
    Returns:
        0: Başarılı
        1: Hata
    """
    try:
        # Dosyanın var olup olmadığını kontrol et
        if not os.path.exists(dosya_yolu):
            print(f"Hata: Dosya bulunamadı: {dosya_yolu}")
            return 1
        
        # Dosya uzantısını kontrol et
        uzanti = os.path.splitext(dosya_yolu)[1].lower()
        if uzanti != '.bmp':
            print("Hata: Bu program sadece .bmp dosyaları için çalışır.")
            return 1
        
        # BMP dosyasını aç
        try:
            img = Image.open(dosya_yolu)
        except Exception as e:
            print(f"Hata: BMP dosyası okunamadı: {e}")
            return 1
        
        # RGB moduna dönüştür (eğer değilse)
        if img.mode != 'RGB':
            img = img.convert('RGB')
        
        eski_genislik = img.width
        yukseklik = img.height
        yeni_genislik = eski_genislik + 1
        
        # Yeni görüntü oluştur (1 piksel daha geniş)
        yeni_img = Image.new('RGB', (yeni_genislik, yukseklik))
        
        # Mavi renk (R:0, G:0, B:255)
        mavi_renk = (0, 0, 255)
        
        # Orijinal pikselleri kopyala
        yeni_img.paste(img, (0, 0))
        
        # Sağ tarafa mavi piksel ekle
        for y in range(yukseklik):
            yeni_img.putpixel((eski_genislik, y), mavi_renk)
        
        # Orijinal dosyayı yedekle
        yedek_dosya_yolu = dosya_yolu + '.yedek'
        if os.path.exists(yedek_dosya_yolu):
            os.remove(yedek_dosya_yolu)
        os.rename(dosya_yolu, yedek_dosya_yolu)
        
        # Yeni görüntüyü kaydet
        yeni_img.save(dosya_yolu, 'BMP')
        
        # Kaynakları temizle
        img.close()
        yeni_img.close()
        
        print("Başarılı!")
        print(f"Dosya genişliği {eski_genislik} pikselden {yeni_genislik} piksele çıkarıldı.")
        print(f"Sağ tarafa mavi (R:0, G:0, B:255) piksel eklendi.")
        print(f"Orijinal dosya yedeklendi: {yedek_dosya_yolu}")
        
        return 0
        
    except Exception as e:
        print(f"Beklenmeyen hata: {e}")
        import traceback
        traceback.print_exc()
        return 1

if __name__ == '__main__':
    if len(sys.argv) < 2:
        print("Kullanım: BmpGenislet.py <bmp_dosya_yolu>")
        print("Örnek: BmpGenislet.py C:\\resim.bmp")
        sys.exit(1)
    
    dosya_yolu = sys.argv[1]
    sys.exit(genislet_bmp(dosya_yolu))

File: BmpGenislet.cs
using System;
using System.IO;
using System.Drawing;
using System.Drawing.Imaging;

/// <summary>
/// BMP dosyasının genişliğini 1 piksel artırır ve sağ tarafa mavi (R:0, G:0, B:255) piksel ekler
/// Windows 7 32-bit için uyumludur
/// </summary>
class Program
{
    static int Main(string[] args)
    {
        try
        {
            // Komut satırından dosya yolunu al
            if (args.Length == 0)
            {
                Console.WriteLine("Kullanım: BmpGenislet.exe <bmp_dosya_yolu>");
                Console.WriteLine("Örnek: BmpGenislet.exe C:\\resim.bmp");
                return 1;
            }

            string dosyaYolu = args[0];

            // Dosyanın var olup olmadığını kontrol et
            if (!File.Exists(dosyaYolu))
            {
                Console.WriteLine("Hata: Dosya bulunamadı: " + dosyaYolu);
                return 1;
            }

            // Dosya uzantısını kontrol et
            string uzanti = Path.GetExtension(dosyaYolu).ToLower();
            if (uzanti != ".bmp")
            {
                Console.WriteLine("Hata: Bu program sadece .bmp dosyaları için çalışır.");
                return 1;
            }

            // BMP dosyasını yükle
            Bitmap orijinalBmp;
            try
            {
                orijinalBmp = new Bitmap(dosyaYolu);
            }
            catch (Exception ex)
            {
                Console.WriteLine("Hata: BMP dosyası okunamadı: " + ex.Message);
                return 1;
            }

            int eskiGenislik = orijinalBmp.Width;
            int yukseklik = orijinalBmp.Height;
            int yeniGenislik = eskiGenislik + 1;

            // Yeni bitmap oluştur (1 piksel daha geniş)
            Bitmap yeniBmp = new Bitmap(yeniGenislik, yukseklik, PixelFormat.Format24bppRgb);

            // Mavi renk (R:0, G:0, B:255)
            Color maviRenk = Color.FromArgb(0, 0, 255);

            // Orijinal pikselleri kopyala ve sağ tarafa mavi piksel ekle
            for (int y = 0; y < yukseklik; y++)
            {
                for (int x = 0; x < eskiGenislik; x++)
                {
                    Color piksel = orijinalBmp.GetPixel(x, y);
                    yeniBmp.SetPixel(x, y, piksel);
                }
                // Sağ tarafa mavi piksel ekle
                yeniBmp.SetPixel(eskiGenislik, y, maviRenk);
            }

            // Orijinal dosyayı yedekle
            string yedekDosyaYolu = dosyaYolu + ".yedek";
            if (File.Exists(yedekDosyaYolu))
            {
                File.Delete(yedekDosyaYolu);
            }
            File.Copy(dosyaYolu, yedekDosyaYolu);

            // Yeni bitmap'i kaydet
            yeniBmp.Save(dosyaYolu, ImageFormat.Bmp);

            // Kaynakları temizle
            orijinalBmp.Dispose();
            yeniBmp.Dispose();

            Console.WriteLine("Başarılı!");
            Console.WriteLine("Dosya genişliği " + eskiGenislik + " pikselden " + yeniGenislik + " piksele çıkarıldı.");
            Console.WriteLine("Sağ tarafa mavi (R:0, G:0, B:255) piksel eklendi.");
            Console.WriteLine("Orijinal dosya yedeklendi: " + yedekDosyaYolu);

            return 0;
        }
        catch (Exception ex)
        {
            Console.WriteLine("Beklenmeyen hata: " + ex.Message);
            Console.WriteLine("Detay: " + ex.StackTrace);
            return 1;
        }
    }
}
